drv/transceivers-server/src/main.rs

```diff
@@ -406,21 +419,56 @@ impl ServerImpl {
                 // be transient (and we'll remove the transceiver on the
                 // next pass through this function).
                 Err(FpgaError::ImplError(e)) => {
-                    match Reg::QSFP::PORT0_STATUS::Encoded::from_u8(e) {
+                    use Reg::QSFP::PORT0_STATUS::Encoded;
+                    match Encoded::from_u8(e) {
                         Some(val) => {
+                            got_nack |= matches!(val, Encoded::I2cAddressNack);
                             ringbuf_entry!(Trace::TemperatureReadError(i, val))
                         }
                         None => {
                             // Error code cannot be decoded
-                            ringbuf_entry!(Trace::InvalidPortStatusError(i, e))
+                            ringbuf_entry!(Trace::InvalidPortStatusError(i, e));
                         }
                     }
                 }
                 Err(e) => {
                     ringbuf_entry!(Trace::TemperatureReadUnexpectedError(i, e));
                 }
             }
+
+            self.consecutive_nacks[i] = if got_nack {
+                self.consecutive_nacks[i].wrapping_add(1)
```

Comment:
It seems like we'd want to saturate instead of wrap, though I _hope_ that's pretty unlikely in any case.

Reply:
Good catch, I failed my coin flip.  Fixed in 5100704ff729d0aaedada5bf6bd1e63d3e81a355